app/src/main/java/ch/epfl/culturequest/storage/LocalStorage.java

```diff
@@ -86,4 +88,31 @@ private void writeImageToStream(Uri uri, Bitmap bitmap) throws IOException {
         }
     }
 
+    /**
+     * Counts the number of selected images in the shared storage.
+     *
+     * @param selection     the selection clause
+     * @param selectionArgs the selection arguments
+     * @return the number of selected images in the shared storage
+     **/
+    public int countSelectedImagesInLocalStorage(String selection, String[] selectionArgs) {
+        // Counts the number of ready images (not pending) in the shared storage
+        try (Cursor cursor = getApplicationContext().getContentResolver().query(
+                contentUri,
+                null,
+                selection,
+                selectionArgs,
+                null)) {
+            return cursor.getCount();
+        }
```

Comment:
Im not sure i understand where we use this function

Reply:
Should be used to check if pending images are in the storage ready to be sent for a scan. This is useful if we decide to do a robust offline mode. If you want some examples of uses, you can check the LocalStorageTest where it is used in tests.